subcommand/inject-connect/command.go

```diff
@@ -323,15 +342,77 @@ func (c *Command) Run(args []string) int {
 		TLSConfig: &tls.Config{GetCertificate: c.getCertificate},
 	}
 
-	c.UI.Info(fmt.Sprintf("Listening on %q...", c.flagListen))
-	if err := server.ListenAndServeTLS("", ""); err != nil {
-		c.UI.Error(fmt.Sprintf("Error listening: %s", err))
-		return 1
+	// channel used for health checks
+	// also check to see if we should enable TLS
+	var healthCh chan struct{}
+	tlsEnabled := os.Getenv("CONSUL_CACERT")
+	consulPort := "8500"
+	if tlsEnabled != "" {
+		consulPort = "8501"
 	}
 
+	if c.flagEnableHealthChecks {
+		go func() {
+			// TODO error handling to close this
+			c.UI.Info(fmt.Sprintf("Listening on %q...", c.flagListen))
+			if err := server.ListenAndServeTLS("", ""); err != nil {
+				c.UI.Error(fmt.Sprintf("Error listening: %s", err))
+			}
+		}()
+
+		syncPeriod, err := time.ParseDuration(c.flagConnectInjectHealthCheckReconcilePeriod)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error parsing health-checks-reconcile-period: %s", err))
+			return 1
+		}
+
+		healthResource := connectinject.HealthCheckResource{
+			Log:          hclog.Default().Named("healthCheckResource"),
+			Clientset:    c.clientset,
+			ClientConfig: api.DefaultConfig(),
```

Comment:
why do we need to set the client config? could we just call `api.DefaultConfig` in the `HealthCheckResource` code?

Reply:
good point! removed all of this!